server/src/main/java/org/elasticsearch/cluster/health/ClusterIndexHealth.java

```diff
@@ -173,19 +258,9 @@ public void writeTo(final StreamOutput out) throws IOException {
         }
     }
 
-    private static final String STATUS = "status";
-    private static final String NUMBER_OF_SHARDS = "number_of_shards";
-    private static final String NUMBER_OF_REPLICAS = "number_of_replicas";
-    private static final String ACTIVE_PRIMARY_SHARDS = "active_primary_shards";
-    private static final String ACTIVE_SHARDS = "active_shards";
-    private static final String RELOCATING_SHARDS = "relocating_shards";
-    private static final String INITIALIZING_SHARDS = "initializing_shards";
-    private static final String UNASSIGNED_SHARDS = "unassigned_shards";
-    private static final String SHARDS = "shards";
-    private static final String PRIMARY_ACTIVE = "primary_active";
-
     @Override
     public XContentBuilder toXContent(final XContentBuilder builder, final Params params) throws IOException {
+        builder.field(INDEX, getIndex());
```

Comment:
Right, I added in PR description:

For discussion in this PR:

- Serialize in XContent object name (in this case 'index' and 'shardId')
  - Pros: Objects can be XContented without level-up name and still have all fields; Easier to parse (including for any other clients)
  - Cons: Non-breaking but API change; A little overhead

So main Pros: each object can be tested for serialization without parent object and can be reused in other classes. But parsing in general will definitely work with a name in Context.

Reply:
ok I had missed that bit in the description. I would not make this change.  I don't completely follow how this makes things easy, it is just one less field that is printed out, but this is a fragment that is usually contained in its own object. Maybe we could rather move the object creation part to this toXContent method then. I would assume that this object is only ever printed out as part of its index object parent, so that might work.

Reply:
So, if I add `startObject()` and `endObject()` to this `toXContent` and change interface to `ToXContentObject`, can I leave `builder.field(INDEX, getIndex())` here? Without this field default XContent serialization/parsing tests will not work.

Reply:
no, the idea would be to add the `startObject(INDEX)` that is now done externally, as well as the corresponding endObject, but that doesn't make this class  a ToXContentObject, it is still a fragment. I would not add the additional index field. Let me know if I am missing anything.

Reply:
I finally got it, thanks. I didn't find a good example of using Fragment which writes `startObject(name)` by itself, has parsing and also has parsing tests. If you can point out, it would be great. So, I wrote this:
https://gist.github.com/Van0SS/9e6e8696a86b5300e55453bd86c9105d
What do you think?

Reply:
It is true that we tend to test mainly requests and responses which print out complete objects. I don't think fragments differ that much from objects though, given that they both implement `ToXContent`. The base test class requires a `ToXContent` and when calling `toXContent` it will wrap the output in an anonymous object when needed (see `XContentHelper#toXContent`). What issues did you encounter? Could you push what you have so I can review?